adc/ade7913/ade7913.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT1176 ADE7913 API
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marcin Baran
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <sys/msg.h>
+
+#include <imxrt-multi.h>
+
+#include "ade7913.h"
+
+
+/* ADE7913 registers address */
+#define ADE7913_IWV          0x0
+#define ADE7913_V1WV         0x1
+#define ADE7913_V2WV         0x2
+#define ADE7913_ADC_CRC      0x4
+#define ADE7913_CTRL_CRC     0x5
+#define ADE7913_CNT_SNAPSHOT 0x7
+#define ADE7913_CONFIG       0x8
+#define ADE7913_STATUS0      0x9
+#define ADE7913_LOCK         0xa
+#define ADE7913_SYNC_SNAP    0xb
+#define ADE7913_COUNTER0     0xc
+#define ADE7913_COUNTER1     0xd
+#define ADE7913_EMI_CTRL     0xd
+#define ADE7913_STATUS1      0xf
+#define ADE7913_TEMPOS       0x18
```

Comment:
By convention, when creating device drivers, we define registers offsets using enum

Reply:
```suggestion
enum { ade7913_iwv = 0, ade7913_v1wv, ade7913_v2wv, ade7913_adc_crc = 4, ade7913_ctrl_crc,
ade7913_cnt_snapshot = 7, ade7913_config, ade7913_status0, ade7913_lock, ade7913_sync_snap,
ade7913_counter0, ade7913_counter1, ade7913_emi_ctrl, ade7913_status1, ade7913_tempos = 24 };
```

Reply:
Changed to enum